engine/src/foundation-legacy.cpp

```diff
@@ -1648,13 +1648,12 @@ static bool save_array_to_stream(void *p_context, MCArrayRef p_array, MCNameRef
         {
             // SN-2014-08-05: [[ Bug 13050 ]] VF_STRING is used for binary as well, so that
             //  we can't use strlen() to get the size of the string.
-            uindex_t t_length;
-            char_t *t_cstring;
-            /* UNCHECKED */ MCStringConvertToNative(t_str_value, t_cstring, t_length);
-			t_stat = ctxt -> stream -> WriteU32(t_length);
+			MCAutoArray<char_t> t_cstring;
```

Comment:
I think you could use MCAutoStringAsNativeChars here rather than AutoArray and ConvertToNative.

Reply:
Ok, I've rewritten to use MCAutoStringRefAsNativeChars now :)